packages/@aws-cdk/aws-ec2/README.md

```diff
@@ -24,32 +24,103 @@ import ec2 = require('@aws-cdk/aws-ec2');
+All default constructs require EC2 instances to be launched inside a VPC, so
 you should generally start by defining a VPC whenever you need to launch
 instances for your project.
 
-Our default `Vpc` class creates a private and public subnet for every
-availability zone. Classes that use the VPC will generally launch instances
-into all private subnets, and provide a parameter called `vpcSubnets` to
-allow you to override the placement. [Read more about
+### Subnet Types
+
+A VPC consists of one or more subnets that instances can be placed into. CDK
+distinguishes three different subnet types:
+
+* **Public** - public subnets connect directly to the Internet using an
+  Internet Gateway. If you want your instances to have a public IP address
+  and be directly reachable from the Internet, you must place them in a
+  public subnet.
+* **Private** - instances in private subnets are not directly routable from the
+  Internet, and connect out to the Internet via a NAT gateway. By default, a
+  NAT gateway is created in every public subnet for maximum availability. Be
+  aware that you will be charged for NAT gateways.
+* **Isolated** - isolated subnets do not route from or to the Internet, and
+  as such do not require NAT gateways. They can only connect to or be
+  connected to from other instances in the same VPC. A default VPC configuration
+  will not include isolated subnets,
+
+A default VPC configuration will create public and private subnets, but not
+isolated subnets. See *Advanced Subnet Configuration* below for information
+on how to change the default subnet configuration.
+
+Classes that use the VPC will "launch instances" (or more accurately, create
+Elastic Network Interfaces) into one or more of the subnets. They all accept
+a property called `subnetSelection` (sometimes called `vpcSubnets`) to allow
+you to select in what subnet to place the ENIs, usually defaulting to
+*private* subnets if the property is omitted.
+
+If you would like to save on the cost of NAT gateways, you can either switch
+to using *isolated* subnets as described in *Advanced Subnet Configuration*,
+or reduce the number of NAT gateways created by setting the `natGateways`
+property to a low number, like 1 or 2. Be aware that this may have
+availability implications for your application.
+
+
+[Read more about
 subnets](https://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Subnets.html).
 
+### Control over availability zones
 
-#### Advanced Subnet Configuration
-If you require the ability to configure subnets the `Vpc` can be
-customized with `SubnetConfiguration` array. This is best explained by an
-example:
+By default, a VPC will spread over at most 3 Availability Zones available to
+it. To change the number of Availability Zones that the VPC will spread over,
+specify the `maxAzs` property when defining it.
 
-```ts
-import ec2 = require('@aws-cdk/aws-ec2');
+How many Availability Zones are available depends on the *region* and
```

Comment:
1. Minor rephrase on 1st line: "The number of availability zones that are available depend on ..."

2. I was aware that AZs are dependent on the region but didn't know they were dependent on the account. Are they?

3. Link to this page would be useful - https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#using-regions-availability-zones-describe

Reply:
2. Yes, although this mostly only happens when new AZs are introduced and there is a whitelisting period.